src/environment.js

```diff
@@ -147,8 +152,7 @@ var Environment = Obj.extend({
                 }
 
                 // Resolve name relative to parentName
-                if (parentName && (name.indexOf("./") == 0 ||
-                                   name.indexOf("../") == 0)) {
```

Comment:
How come you removed the check for a relative path? We don't want to resolve every single template. Wouldn't that break all current templates? A normal template like `foo.html` should resolve to the root folder. I don't see any other changes related to this.